File: Laravel/app/Http/Requests/Api/Auth/SignUpRequest.php
<?php

namespace App\Http\Requests\Api\Auth;

use App\Http\Requests\ApiRequest;

class SignUpRequest extends ApiRequest
{
    public function rules(): array
    {
        return [
            'login' => 'required|string|unique:users',
            'email' => 'nullable|string|email|max:255|unique:users',
            'password' => 'required|string|min:6|max:255|confirmed',
            'password_confirmation' => 'required|string|min:6|max:255',
        ];
    }
}

File: PSB/Api/Request/SignUpRequest.cs
using System.Text.Json.Serialization;

namespace PSB.Api.Request
{
    public class SignUpRequest(string login, string? email, string password, string passwordConfirmation)
    {
        [JsonPropertyName("login")] public string Login { get; set; } = login;
        [JsonPropertyName("email")] public string? Email { get; set; } = email;
        [JsonPropertyName("password")] public string Password { get; set; } = password;
        [JsonPropertyName("password_confirmation")] public string PasswordConfirmation { get; set; } = passwordConfirmation;
    }
}
